test/_init_tests.js

```diff
@@ -207,6 +207,49 @@ describe.configure = function() {
   return new TestConfig(describe);
 };
 
+function enableYield(fn, message, runnable) {
+  if (!runnable || !runnable.constructor
+      || runnable.constructor.name !== 'GeneratorFunction') {
+    return fn(message, runnable);
+  }
+  return fn(message, done => {
+    const runner = (iterator, result) => {
```

Comment:
just want to understand the benefit better:

> Promises handle async (through assimilation)

Sorry, what does that mean? :-)

> Yielded promises can reject, going into the generators throw path

This is not a problem. I should have added anyway.

> We avoid using done #8966

I handled this by using a try-catch block. But in your impl, there's also a try-catch, is that there for the same reason?

> Less closures

If I count correctly, all the 3 functions are closures `step`, `_next`, `_throw`. Not sure how it is less than the `done` impl.

Also, I still think there is value to support setTimeout. We will add boilerplate code to turn a setTimeout to promise.